INT: use text with aliases in Type Info intention

Kobzol opened the pull request:
I replaced `toString()` with `insertionSafeTextWithAliases` to show aliases in the `Type Info` hint. I didn't find any tests for this, so I'm not sure if this is enough. @Undin thoughts?

Fixes https://github.com/intellij-rust/intellij-rust/issues/4366

Undin commented on src/main/kotlin/org/rust/ide/hints/RsExpressionTypeProvider.kt:
```diff
@@ -35,7 +36,7 @@ class RsExpressionTypeProvider : ExpressionTypeProvider<PsiElement>() {
     override fun getInformationHint(element: PsiElement): String {
         val type = getType(element)
-        return type.toString().escaped
+        return type.insertionSafeTextWithAliases.escaped
```
You should use `textWithAliasNames` instead of `insertionSafeTextWithAliases ` (I know that it's marked a `TestOnly`. It's time to drop this annotation) not to change the previous style of how we render unknown types. `insertionSafeTextWithAliases` is used as text for PSI element creation in intentions/quick fixes/refactorings/etc.

Kobzol commented:
I added a basic test. The provider returned escaped text, so I introduced a method which returns the raw text for easier testing. Let me know if the test structures looks OK. In that case I will add more tests.

Undin commented on src/test/kotlin/org/rust/ide/hints/RsExpressionTypeProviderTest.kt:
```diff
@@ -0,0 +1,46 @@
+
+        val provider = RsExpressionTypeProvider()
+        val expressions = provider.getExpressionsAt(element)
+        assertEquals(type, expressions.joinToString(", ") { provider.getInformationHintUnescaped(it) })
```
Let's better use `getInformationHint` to ensure that we don't forget to unescape something

Kobzol replied:
Ok. I didn't want to use it because then if you have a test failure you see something like "Expected: & lt;Something;& gt". But it's not a big deal.

Kobzol commented:
Added more tests.

Undin commented on src/test/kotlin/org/rust/ide/hints/RsExpressionTypeProviderTest.kt:
```diff
@@ -0,0 +1,84 @@
+import org.rust.RsTestBase
+import org.rust.WithStdlibRustProjectDescriptor
+import org.rust.openapiext.escaped
+import kotlin.test.assertEquals
```
Could you remove unused imports?

Kobzol commented:
Removed unused imports and squashed commits.
